### PR TITLE
Add the ability to perform a dependency check scan.

### PR DESCRIPTION
**Pull request by jcouture**
This PR is all about dependency check to find publicly disclosed vulnerabilities:

* Integrates a Gradle task to perform dependency check scans.
* Adds a GitHub Actions Workflow to weekly trigger the dependency check and upload the resulting SARIF (Static Analysis Results Interchange Format) file to GitHub Code Scanning functionality.

:v:

### CODE REVIEW

File: .github/workflows/dependency-check.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      - uses: actions/cache@v3
+        with:
+          path: ~/.gradle
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - uses: actions/setup-java@v3
+        with:
+          distribution: "temurin"
```
Comment (marcantoinefortier) [inline .github/workflows/dependency-check.yaml:23]: Is it necessary to specific a distribution here? We don't seem to need it in https://github.com/mirego/kmp-boilerplate/blob/main/.github/workflows/ci.yaml#L32 👀

File: build.gradle.kts [hunk 2]
```diff
@@ -7,8 +7,19 @@ plugins {
...
     delete(rootProject.buildDir)
 }
+
+dependencyCheck {
+    formats = listOf("HTML", "SARIF")
+    outputDirectory = "reports"
+
+    analyzers.assemblyEnabled = false
+    analyzers.experimentalEnabled = true
+    analyzers.knownExploitedEnabled = true
```
Comment (marcantoinefortier) [inline build.gradle.kts:24]:
```suggestion
    analyzers.apply {
        assemblyEnabled = false
        experimentalEnabled = true
        knownExploitedEnabled = true
    }
```

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -28,6 +28,7 @@ kotlin-multiplatform = { id = "org.jetbrains.kotlin.multiplatform", version.ref
 kotlin-native-cocoapods = { id = "org.jetbrains.kotlin.native.cocoapods", version.ref = "kotlin" }
 serialization = { id = "org.jetbrains.kotlin.plugin.serialization", version.ref = "kotlin" }
 ktlint = { id = "org.jlleitschuh.gradle.ktlint", version.ref = "ktlint" }
+owasp-dependencycheck = { id = "org.owasp.dependencycheck", version = "8.4.0" }
```
Comment (marcantoinefortier) [inline gradle/libs.versions.toml:31]: Let's place it in alphabetical order (above `serialization`) to keep it constant 🙏🏻

**Review by marcantoinefortier — APPROVED** [review]

**Comment by marcantoinefortier** [comment]
I think we should also write an entry in the README.md to document this new feature 🙂

File: .github/workflows/dependency-check.yaml (hunk 1 above)
Comment (jcouture) [reply .github/workflows/dependency-check.yaml:23]: I opted for a more recent version of the setup-java action, which requires the distribution to be set. We could potentially update the CI workflow and even use the same distribution in the .tools-version file.

Temurin is the recommended distribution of the OpenJDK according to this fine reference:
https://whichjdk.com

File: gradle/libs.versions.toml (hunk 3 above)
Comment (jcouture) [reply gradle/libs.versions.toml:31]: It was already not in alphabetical order, afaik?

`serialization` being before `ktlint`... just sayin' 😏

File: .github/workflows/dependency-check.yaml (hunk 1 above)
Comment (marcantoinefortier) [reply .github/workflows/dependency-check.yaml:23]: Great! Thanks for the reference 🙌🏻 Let's update our other github actions and our `.tools-version` then 🙂

File: gradle/libs.versions.toml (hunk 3 above)
Comment (marcantoinefortier) [reply gradle/libs.versions.toml:31]: You're absolutely right 🤦‍♂️ Let's fix ktlint too then 😅

**Review by marcantoinefortier — APPROVED** [review]
Thanks for the README info!

**Pull request merged**